modules/catalog/src/main/java/org/apache/ignite/internal/catalog/events/CatalogEventParameters.java

```diff
@@ -23,12 +23,24 @@
  * Base class for Catalog event parameters.
  */
 public abstract class CatalogEventParameters extends EventParameters {
+    private final int catalogVersion;
+
     /**
      * Constructor.
      *
      * @param causalityToken Causality token.
+     * @param catalogVersion Catalog version.
      */
-    public CatalogEventParameters(long causalityToken) {
+    public CatalogEventParameters(long causalityToken, int catalogVersion) {
```

Comment:
why version of the catalog can't be causality token?

Reply:
I was involved in it. Right now we only have meta-storage revision as a causality token. If we use two different entities, calling it the same name, it will lead to confusion. It's easier and safer to migrate meta-storage revision into a catalog version later with a single commit